modules.d/95nvmf/module-setup.sh

```diff
@@ -25,17 +25,27 @@ check() {
         [[ $trtype == "fc" ]] || [[ $trtype == "tcp" ]] || [[ $trtype == "rdma" ]]
     }
 
+    has_nbft() {
+        local f found=
+        for f in /sys/firmware/acpi/tables/NBFT*; do
+            [ -f "$f" ] || continue
```

Comment:
This checks only for regular files. Is that intentional?

You could check for `-e` (existence) instead.

Reply:
ok.

Reply:
Right, folders, I didn't think of that - but It might be symlink or other file-like type as well f.e.; but if only regular files are expected/accepted, I'm fine with that.

Reply:
Yes, only regular files should be accepted. 

Note that `[` (aka `test`) dereferences symlinks for all flags except `-h` and `-L`. Thus if `/sys/firmware/acpi/tables/NBFT*` was a symlink to some existing regular file, this test wouldn't skip it, which looks wrong. But this case would indicate a severely broken kernel. I believe we can ignore it. In the worst case, we'd include the NBFT code in the initramfs on a broken system.